tests/src/testnonlinearelasticityneohooke.cpp

```diff
@@ -28,8 +28,8 @@ int main(int argc, char** argv) {
   t.subTest(NonLinearElasticityLoadControlNRandTR<Grids::Yasp>(matNH1));
   t.subTest(NonLinearElasticityLoadControlNRandTR<Grids::IgaSurfaceIn2D>(matNH1));
 
-  autoDiffTest<2>(t, planeStressMat1, " nu != 0");
-  autoDiffTest<2>(t, planeStressMat2, " nu = 0");
+  autoDiffTest<2>(t, planeStressMat1, " nu != 0", 1e-7);
```

Comment:
This tolerance isn't always necessary but only for certain cases depending on the randomness in the element geometry and the displacement vector (see [comment](https://github.com/ikarus-project/ikarus/pull/329#issuecomment-2416967885)). I am not sure if this is good enough or it could be resolved in a better way? May be fixing the random displacement vector in `checkFESByAutoDiff` such that this tolerance can be reduced further? We can also fix the tolerance while working on the Issue #211.

Reply:
Okay then I leave the tolerance here as such and we can refactor `isApproxSame` while perhaps working on #211